src/libsyntax/parse/parser.rs

```diff
@@ -2246,6 +2258,15 @@ impl<'a> Parser<'a> {
                     hi = path.span;
                     return Ok(self.mk_expr(lo.to(hi), ExprKind::Path(Some(qself), path), attrs));
                 }
+                if syntax_pos::hygiene::default_edition() >= Edition::Edition2018 &&
```

Comment:
You probably want `self.token.span.edition() >= Edition::Edition2018` here instead of default edition in case `async` comes from a macro defined with other edition.

Reply:
Do you mean `self.span.edition()`? `Token` doesn't have a `span` field.

Reply:
Yes, sorry (`self.span` is the span of `self.token`).